pkg/controllers/dynakube/dynakube_controller.go

```diff
@@ -266,6 +284,7 @@ func (controller *Controller) reconcileDynaKube(ctx context.Context, dynakube *d
 		return err
 	}
 
+	log.Info("start reconciling connection info")
 	err = controller.reconcileConnectionInfo(ctx, dynakube, dynatraceClient)
```

Comment:
**Main Suggestion** 👇 

I would just add a `controller.updateDynakubeStatus(ctx, dynaKube)` here, instead of all this convoluted logic, if we want to further relay on timely status refreshes, then we should start updating the status as soon as we can,

restarting the whole `Reconcile` does the same, but with many more steps

Reply:
Actually, I was working under the assumption, that a Reconile restart is the proper way to do this. If just writing back the updated status is sufficient, I'm more than happy to do it like that.